app/uk/gov/hmrc/gform/des/DesConnector.scala

```diff
@@ -36,18 +37,73 @@ class DesConnector(wSHttp: WSHttp, baseUrl: String, desConfig: DesConnectorConfi
        "isAnAgent": false
       }""") //TODO add in actual regime we are looking for
 
-  def lookup(utr: String)(implicit hc: HeaderCarrier, ex: ExecutionContext): Future[AddressDes] = {
+  def lookupAddress(utr: String)(implicit hc: HeaderCarrier, ex: ExecutionContext): Future[AddressDes] = {
     implicit val hc = HeaderCarrier(
       extraHeaders = Seq("Environment" -> desConfig.environment),
       authorization = Some(Authorization(s"Bearer ${desConfig.authorizationToken}")))
     Logger.info(s"Des lookup, UTR: '$utr', ${loggingHelpers.cleanHeaderCarrierHeader(hc)}")
     wSHttp.POST[JsValue, AddressDes](s"$baseUrl${desConfig.basePath}/registration/organisation/utr/$utr", lookupJson)
   }
 
+  def lookupTaxPeriod(idType: String, idNumber: String, regimeType: String)(
+    implicit hc: HeaderCarrier,
+    ex: ExecutionContext): Future[Obligation] = {
+    implicit val hc = HeaderCarrier(
+      extraHeaders = Seq("Environment" -> desConfig.environment),
+      authorization = Some(Authorization(desConfig.authorizationToken)))
+    Logger.info(
+      s"Des lookup, Tax Periods: '$idType, $idNumber, $regimeType', ${loggingHelpers.cleanHeaderCarrierHeader(hc)}")
+    wSHttp.GET[Obligation](
+      s"$baseUrl${desConfig.basePath}/enterprise/obligation-data/$idType/$idNumber/$regimeType?status=O") recoverWith {
+      case e: JsValidationException => Future(Obligation(List()))
```

Comment:
I think that we should only accept what the obligations API is supposed to return in the case of no obligations as an empty list of obligations.  Otherwise it ought to be handled as an upstream error